core/controllers/tasks_test.py

```diff
@@ -0,0 +1,340 @@
+                self.process_and_flush_pending_tasks()
+
+                #check that user B recieved message.
+                messages = self.mail_stub.get_sent_messages(
+                    to=self.USER_B_EMAIL)
+                self.assertEqual(len(messages), 1)
+
+                #check that user B recieved correct message.
+                expected_message = (
+                    'Hi userB,\nuserA has submitted a new suggestion'
+                    ' for your Oppia exploration, "Title".\nYou can'
+                    ' accept or reject this suggestion by visiting'
+                    ' the feedback page for your exploration.\n\nTha'
+                    'nks!\n- The Oppia Team\n\nYou can change your'
+                    ' email preferences via the Preferences page.')
+                self.assertEqual(messages[0].body.decode(), expected_message)
+
+    def test_InstantFeedbackMessageEmailHandler(self):
+        """Tests Instant feedback message handler."""
+        with self.can_send_feedback_email_ctx, self.can_send_emails_ctx:
+            feedback_services.create_thread(
+                feconf.ENTITY_TYPE_EXPLORATION, self.exploration.id,
+                self.user_id_a, 'a subject', 'some text')
+            threadlist = feedback_services.get_all_threads(
+                feconf.ENTITY_TYPE_EXPLORATION, self.exploration.id, False)
+            thread_id = threadlist[0].id
+            #create reply message.
+            feedback_services.create_message(
+                thread_id, self.user_id_b, None, None, 'user b message')
+            #get all messages in the thread.
+            messages = feedback_services.get_messages(thread_id)
+            #make sure there are only 2 messages in thread.
+            self.assertEqual(len(messages), 2)
+
+            #ensure that user A has no emails sent yet.
+            messages = self.mail_stub.get_sent_messages(
+                to=self.USER_A_EMAIL)
+            self.assertEqual(len(messages), 0)
+
+            #invoke InstantFeedbackMessageEmail which sends.
+            #InstantFeedbackMessage.
+            self.process_and_flush_pending_tasks()
+
+            #ensure that user A has an email sent now.
+            messages = self.mail_stub.get_sent_messages(
+                to=self.USER_A_EMAIL)
+            self.assertEqual(len(messages), 1)
+
+            #ensure that user A has right email sent to them.
+            expected_message = (
+                'Hi userA,\n\nNew update to thread "a subject" on'
+                ' Title:\n- userB: user b message\n(You received'
+                ' this message because you are a participant in'
+                ' this thread.)\n\nBest wishes,\nThe Oppia'
+                ' team\n\nYou can change your email preferences'
+                ' via the Preferences page.')
+            self.assertEqual(messages[0].body.decode(), expected_message)
+
+    def test_FeedbackThreadStatusChangeEmailHandler(self):
```

Comment:
Just a note, this should actually be test_feedback_thread_status_change_email_handler (or, better, something that describes what is actually being tested, e.g. test_email_is_sent_when_thread_status_changes). But basically it should all be snake case, like in other files. The reason there are occasional lowerCamelCase things in tests is for stuff like setUp and tearDown which is an artifact of the corresponding python stdlib, but otherwise we use snake_case for method/function names in python.

Reply:
Oops, my bad! I confused the conventions we use. Please follow what @seanlip said!

Reply:
<img width="644" alt="screen shot 2018-11-08 at 11 21 24 am" src="https://user-images.githubusercontent.com/39934918/48222573-ba267380-e349-11e8-9740-75946abbd061.png">

This is the screenshot from generating coverage report using: bash scripts/ run_backend_tests.sh --test_target = core.controllers.tasks_test --generate_coverage_report

Reply:
Done.